File: objectiveai-js/src/functions/remote.ts
import z from "zod";
import { convert, type JSONSchema } from "../json_schema";

export const RemoteSchema = z.enum(["github", "filesystem"]).describe(
  "The remote source hosting the repository.",
);
export type Remote = z.infer<typeof RemoteSchema>;
export const RemoteJsonSchema: JSONSchema = convert(RemoteSchema);

File: objectiveai-rs/src/functions/remote.rs
//! Remote source types for function and profile hosting.

use serde::{Deserialize, Serialize};
use std::fmt;

/// The remote source where a function or profile is hosted.
#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq, Eq, Hash)]
#[serde(rename_all = "snake_case")]
pub enum Remote {
    /// GitHub repository.
    Github,
    /// Local filesystem.
    Filesystem,
}

impl fmt::Display for Remote {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        match self {
            Remote::Github => write!(f, "github"),
            Remote::Filesystem => write!(f, "filesystem"),
        }
    }
}
